test/integration/parallel.spec.js

```diff
@@ -18,4 +18,42 @@ describe('parallel run', () => {
     assert.strictEqual(result.stats.failures, 1);
     assert.strictEqual(result.stats.passes, 2);
   });
+
+  it('should correctly set worker ids for each process', async () => {
+    const result = await runMochaJSONAsync('parallel/testworkerid3.mjs', [
+      '--parallel',
+      '--jobs',
+      '2',
+      require.resolve('./fixtures/parallel/testworkerid1.mjs'),
+      require.resolve('./fixtures/parallel/testworkerid2.mjs')
+    ]);
+    // test 1 always runs on worker 0
+    assert.strictEqual(
+      result.tests.filter(
+        test =>
+          test.fullTitle === 'test1 should fail' &&
+          test.err.message === 'MOCHA_WORKER_ID=0'
+      ).length,
+      1
+    );
+    // test 2 always runs on worker 1
+    assert.strictEqual(
+      result.tests.filter(
+        test =>
+          test.fullTitle === 'test2 should fail' &&
+          test.err.message === 'MOCHA_WORKER_ID=1'
+      ).length,
+      1
+    );
+    // test 3 may run on either worker 0 or 1 (as jobs = 2)
+    assert.strictEqual(
```

Comment:
When a worker is reused after having finished its test file, it keeps its ID, right? Is this a good solution? Does it help eg. users who want to connect to a different database with each worker/test file?

Reply:
I think the main use case is to avoid sharing resources that cannot be used in parallel (ports or DB for example), so reusing resources in each files that are run by the same worker should be fine.

There might be some use cases for having an id per file, but it would be a different use case (in particular that use case would also make sense for non parallel mode, while the current PRs is specifically to solve a problem in parallel mode)

Reply:
Do you know how jest handles this ID when re-using a worker?

Do you know by chance how workerpool cleans up a process before re-use? I'm refering eg. to CJS require cache or EMS import cache?

Reply:
Sorry I missed your answer earlier. The way mocha uses it, workerpool launches processes using child_process.fork, which means it creating new full node processes. 

Then I don't think it's doing anything fancy. The mocha workerpool will run this file https://github.com/mochajs/mocha/blob/master/lib/nodejs/worker.js then the main node process will send the file to run and the child process will process it, then report result and get another file to run. Unless mocha does some special clean up, I think the files running in the same processes will share everything (cache & all)

Reply:
From what I could understand from a quick look of their code, jest seems to work similarly, though they use their own worker pool implementation, which seems slightly more complex than workerpool, so I'm not 100% sure I understood everything :)

Reply:
Thanks for your explanations. Let's go your way then, which means an unique ID for each worker,